modules/lang/fortran/autoload.el

```
@@ -115,3 +115,10 @@ or gfortran, depending on what feature flags are set."
   "Test the current project using fpm."
   (interactive)
   (compile "fpm test"))
+
+;;;###autoload
+(defun +fortran/fpm-open-project-toml ()
+  "Open the TOML file associated with the current project."
+  (interactive)
+  (when-let ((root (doom-project-root)))
+    (find-file (file-name-concat root "fpm.toml"))))
```

Silently does nothing when not in an FPM-based project.

---

To make it more versatile and improve its feedback, please change it to:

```elisp
(defun +fortran/fpm-open-project-toml (project-root)
  "Open fpm.toml at PROJECT-ROOT (defaults to the open project)."
  (interactive (list (doom-project-root)))
  (let ((file (file-name-concat project-root "fpm.toml")))
    (if (file-exists-p file)
        (find-file file)
      (user-error "No fpm.toml found at project root (%s)"
                  (abbreviate-file-name project-root)))))
```

---

That would be much better yeah. Thanks for reminding me about `user-error`!